devito/operations/interpolators.py

```diff
@@ -388,3 +389,520 @@ def callback():
+
+
+class SincInterpolator(GenericInterpolator):
+    """
+    Concrete implementation of GenericInterpolator implementing a Sinc interpolation
+    scheme, using a method that is windowed by a Kaiser Function.
+    Parameters
+    ----------
+    sfunction: The SparseFunction that this Interpolator operates on.
+    """
+
+    def __init__(self, sfunction):
+        self.sfunction = sfunction
+        self.r = 4
+
+    @property
+    def grid(self):
+        return self.sfunction.grid
+
+    def _sinc_equation(self, expr, position, dim_pos, idx_subs, idx2d, idx3d=None):
+        """
+        Generate the basic sinc equation.
+        Parameters
+        ----------
+        expr : expr-like
+            Input expression to interpolate.
+        position: List
+            Elements which represents the position of the interpolation point
+            multiplied by their respective coefficients.
+        dim_pos: Symbol
+            Symbol representing the dimension wich will be used to form the sinc
+            equations.
+        idx_subs: dict
+            Structure responsible for mapping the order of substitution of dimensions
+            in expr.
+        idx_2d: Integer
+            Defines which iteration of second dimension is being used.
+        idx3d: Integer, optional
+            defines which iteration of third dimension is being used,
+            if it is a 3D interpolation.
+        """
+
+        # Defining the symbolic function responsible for accessing
+        # the pre-computed coefficient value
+        def sincKernel(args):
+            coeffs = sympy.Function(name="acessCoeffs")(args)
+            return coeffs
+
+        npoints = self.r*2
+
+        pos = sympy.Matrix(position)
+
+        dim_pos = sympy.Matrix(dim_pos)
+
+        p = [expr.xreplace(subs) for subs in idx_subs]
+        p = sympy.Matrix(p)
+
+        index_opt = [idx3d, idx2d]
+        index_opt = [b for b in index_opt if b is not None]
+        arg_kernel = [pos[ii] - dim for ii, dim in enumerate(dim_pos)]
+
+        kernel = [sincKernel(arg) for arg, index in zip(arg_kernel, enumerate(index_opt))]
+
+        kernel = np.prod(kernel)
+        kernel = [kernel*sincKernel(arg_kernel[-1]) for ii in range(npoints)]
+
+        points = [point*k.subs(idx) for point, k, idx in zip(p, kernel, idx_subs)]
+
+        return points
+
+    def _sinc_equations2D(self, expr, idx_subs, dim_pos=None):
+        """
+        Generate equations responsible for 2D sinc interpolation's computation.
+        """
+        # Gets the list containing the symbols px and py
+        pos = self.sfunction._point_symbols
+
+        # n = number of neighbor points
+        n = self.r*2
+
+        # Generation of equations
+        eqs = [self._sinc_equation(expr, pos, dim_pos, idx_subs[ii*n:(ii+1)*n], idx2d=ii)
+               for ii in range(n)]
+
+        return eqs
+
+    def _sinc_equations3D(self, expr, idx_subs, dim_pos=None):
```

Comment:
Again, can these be consolidated into `_sinc_equationsND()`? Seems like a decent amount of lines are the same and would be nice to have a generalization to N-D